unified_planning/solvers/pddl_solver.py

```diff
@@ -79,14 +85,39 @@ def solve(self, problem: 'up.model.Problem',
             w.write_domain(domanin_filename)
             w.write_problem(problem_filename)
             cmd = self._get_cmd(domanin_filename, problem_filename, plan_filename)
-            try:
-                res = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=timeout)
-                logs.append(up.solvers.results.LogMessage(up.solvers.results.INFO, res.stdout.decode()))
-                logs.append(up.solvers.results.LogMessage(up.solvers.results.ERROR, res.stderr.decode()))
-            except subprocess.TimeoutExpired:
-                return PlanGenerationResult(up.solvers.results.TIMEOUT, plan, planner_name=self.name())
+
+            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            timer = Timer(timeout, proc.kill) # type: ignore #NOTE Timer doc does not say it can accept a None timeout, but if None is given no problems are raised.
+            timer.start()
+            proc_out: str = ''
+            proc_err: str = ''
+            if out is not None:
+                while True:
+                    r, _, _ = select.select([proc.stdout, proc.stderr], [], [])
+                    for readable_stream in r:
+                        if readable_stream == proc.stdout:
+                            out_in_bytes = proc.stdout.read(SIZE_TO_READ)
+                            out_str = out_in_bytes.decode()
+                            out.write(out_str)
+                            proc_out = f'{proc_out}{out_str}'
+                        elif readable_stream == proc.stderr:
+                            err_in_bytes = proc.stderr.read(SIZE_TO_READ)
+                            err_str = err_in_bytes.decode()
+                            out.write(err_str)
+                            proc_err = f'{proc_err}{err_str}'
```

Comment:
KISS
```suggestion
                        out_in_bytes = readable_stream.read(SIZE_TO_READ)
                        out_str = out_in_bytes.decode()
                        out.write(out_str)
                        if readable_stream == proc.stdout:
                            proc_out.append(out_str)
                        else:
                            proc_err.append(err_str)
```

Then at the end, you can get the output and error string as `"".join(proc_out)`